Tests/FunctionalTests/TestDiscoveryTests.swift

```diff
@@ -12,159 +12,156 @@
 
 import Basics
 import PackageModel
+import SPMBuildCore
 import _InternalTestSupport
-import XCTest
+import Testing
+import struct Foundation.UUID
 
-final class TestDiscoveryTests: XCTestCase {
-    func testBuild() async throws {
+@Suite
+struct TestDiscoveryTests {
+
+    @Test(arguments: [BuildSystemProvider.Kind.native, .swiftbuild])
+    func build(_ buildSystem: BuildSystemProvider.Kind) async throws {
         try await fixture(name: "Miscellaneous/TestDiscovery/Simple") { fixturePath in
-            let (stdout, _) = try await executeSwiftBuild(fixturePath)
+            let (stdout, _) = try await executeSwiftBuild(fixturePath, buildSystem: buildSystem)
             // in "swift build" build output goes to stdout
-            XCTAssertMatch(stdout, .contains("Build complete!"))
+            #expect(stdout.contains("Build complete!"))
         }
     }
 
-    func testDiscovery() async throws {
+    @Test(arguments: [BuildSystemProvider.Kind.native, .swiftbuild])
+    func discovery(_ buildSystem: BuildSystemProvider.Kind) async throws {
         try await fixture(name: "Miscellaneous/TestDiscovery/Simple") { fixturePath in
-            let (stdout, stderr) = try await executeSwiftTest(fixturePath)
+            let (stdout, stderr) = try await executeSwiftTest(fixturePath, buildSystem: buildSystem)
             // in "swift test" build output goes to stderr
-            XCTAssertMatch(stderr, .contains("Build complete!"))
+            #expect(stderr.contains("Build complete!"))
             // in "swift test" test output goes to stdout
-            XCTAssertMatch(stdout, .contains("Executed 3 tests"))
+            #expect(stdout.contains("Executed 3 tests"))
         }
     }
 
-    func testNonStandardName() async throws {
+    // FIXME: Fix --build-system swiftbuild link file list quoting on non-Darwin
```

Comment:
question: do we have a GitHub issue for this? or is to plan to address this FIXME before merge?  Can we create a GitHub issue and add a `.bug()` trait that references said issue?